### ISSUE TITLE
RCS1219: Incorrect fix for `list.Count > list.Distinct().Count()`

### ISSUE THREAD

**Issue by JoshuaLight** [JosefPihrt/Roslynator#729]
**Product and Version Used**: `Roslynator.Analyzers 3.0.0`

**Steps to Reproduce**:
```cs
private static bool ContainsDuplicates<T>(this IList<T> self) =>
    self.Count > self.Distinct().Count();
```

**Actual Behavior**:
```cs
private static bool ContainsDuplicates<T>(this IList<T> self) =>
    !self.Distinct().Skip(self.Count - 1).Any();
```

**Expected Behavior**:
```csharp
private static bool ContainsDuplicates<T>(this IList<T> self) =>
    self.Skip(self.Distinct().Count()).Any();
```

**Explanation**:
The problem with `!self.Distinct().Skip(self.Count - 1).Any()` is that for empty list it equals to `true`:
1) `[].Distinct()` -> `[]`;
2) `[].Skip(0 - 1)` -> `[]` (From [docs](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.skip?view=netcore-3.1): _"If `count` is less than or equal to zero, all elements of source are yielded"_);
3) `[].Any()` -> `false`;
4) `!false` -> `true`.

However, the initial expression `self.Count > self.Distinct().Count()` would be `false`: `0 > 0` -> `false`.

**Comment by JosefPihrt** [comment]
Thanks for this report.

I did some analysis and I found out that right-side expression must be constant value `>= 0` (or `> 0`).

```cs
// i must be constant value >= 0
items.Count() > i  /* >>> */ items.Skip(i).Any()
items.Count() <= i /* >>> */ !items.Skip(i).Any()
```

```cs
// i must be constant value > 0
items.Count() >= i /* >>> */ items.Skip(i - 1).Any()
items.Count() < i  /* >>> */ !items.Skip(i - 1).Any()
```

Original intent of this analyzer was to prevent enumerating entire sequence if possible.
(Which by the way means that your proposal (even if semantically correct) does not match this intent because `Distinct`
must enumerate all items to filter unique items.)

The problem with this analyzer was that code fix itself can be considered as too clumsy to be actually valuable and because of that the analyzer is disabled by default.

And now with your findings it is clear that applicability of this analyzer is very limited.

So I think that best solution would be to remove this analyzer from Roslynator.
